apollo-router/src/services/supergraph_service.rs

```diff
@@ -242,18 +258,32 @@ where
 
 async fn plan_query(
     mut planning: CachingQueryPlanner<BridgeQueryPlanner>,
-    body: &graphql::Request,
+    compiler: Option<Arc<Mutex<ApolloCompiler>>>,
+    operation_name: Option<String>,
     context: Context,
+    schema: Arc<Schema>,
+    query_str: String,
 ) -> Result<QueryPlannerResponse, CacheResolverError> {
+    let compiler = match compiler {
+        None =>
+        // TODO[igni]: no
```

Comment:
this is fixable if we made APQ and query analysis work on a graphql request object instead of a supergraph request in router service

Reply:
I just tried and even then, dozens of tests fail. The only way around it is in generating a compiler in every place we create a supergraph request, and at the same time break the public API. I'd say we should keep that fix in there for now, and reasses when we make the compiler public

Reply:
Maybe even a fast follow once this is merged. I'm guessing the whole optional things is because of APQ being in the wrong place.

Could APQ even be moved before this PR merges?

Reply:
I'd prefer it as a fast follow, because this PR is already getting big enough and touches a lot of files. APQ does not need to move, it's mainly about which types it works on.
The alternative would be to just create a compiler directly in the supergraph request builder. It would not have the schema information, but that would be enough for tests